next/components/forms/segments/AccountSections/MyApplicationsSection/MyApplicationsList.tsx

```diff
@@ -40,58 +38,47 @@ export const getDraftApplications = async (
     undefined,
     variantToStates,
     undefined,
-    { accessToken: 'always', accessTokenSsrReq },
+    accessTokenSsrReq ? { accessToken: 'always', accessTokenSsrReq } : { accessToken: 'always' },
   )
   return response.data
 }
 
 type MyApplicationsListProps = {
   variant: ApplicationsListVariant
+  applications?: GetFormsResponseDto
+  refetchApplicationsCount: () => Promise<void>
 }
 
-const MyApplicationsList = ({ variant }: MyApplicationsListProps) => {
-  const { t } = useTranslation('account')
-  const [openSnackbarError] = useSnackbar({ variant: 'error' })
-
+const MyApplicationsList = ({
+  variant,
+  applications,
+  refetchApplicationsCount,
+}: MyApplicationsListProps) => {
   const router = useRouter()
   const currentPage = parseInt(router.query.strana as string, 10) || 1
 
-  const {
-    data,
-    isError,
-    isLoading: isQueryFirstLoad,
-    refetch,
-    isRefetching,
-  } = useQuery({
-    queryKey: [`myApplications_${variant}`, currentPage],
-    queryFn: () => getDraftApplications(variant, currentPage),
-  })
-
-  const totalPagesCount = data?.countPages ?? 0
-  // error is shown through snackbar - since we don't have error state we keep the list stuck on loading
-  // we trigger refetch on concept delete - this way it trigger loading state for the user
-  // better to have instant feedback than to be confused whether the request went through
-  const isLoading = isQueryFirstLoad || isRefetching || (isError && !data?.items?.length)
+  const { refreshData } = useRefreshServerSideProps(applications)
 
-  const forms: Array<GetFormResponseDto | null> = isLoading
-    ? (Array.from({ length: 10 }).fill(null) as Array<null>)
-    : data?.items ?? ([] as Array<GetFormResponseDto>)
+  const refreshListData = async () => {
+    await refetchApplicationsCount()
+    await refreshData()
+  }
 
-  useEffect(() => {
-    if (isError) openSnackbarError(t('account_section_applications.error'))
-    // TODO openSnackbarError is a new reference on every render - fix dependencies once this hooks is fixed
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isError])
+  const totalPagesCount = applications?.countPages ?? 0
 
   return (
     <div className="m-auto w-full max-w-screen-lg">
-      {forms.length > 0 ? (
+      {applications?.items.length && applications.items.length > 0 ? (
```

Comment:
perhaps a thing of preference, but imo this is needlessly verbose, you can safely do just

```suggestion
      {applications?.items.length ? (
```